provider/ec2/environ.go

```diff
@@ -214,31 +214,23 @@ func (e *environ) Bootstrap(ctx environs.BootstrapContext, callCtx context.Provi
 	// We are going to take a look at the Bootstrap constraints and see if we have to make an instance profile
 	if args.BootstrapConstraints.HasInstanceRole() &&
 		*args.BootstrapConstraints.InstanceRole == awsInstanceProfileAutoCreateVal {
-
-		// Added by tlm on 07/10/2021 (This will be removed very shortly).
-
-		// If the user has asked us to automatically create the instance
-		// profile for them we will return a not supported for now. This is till
-		// the remainder of the AWS work is complete and we have concrete role
-		// policies to use.
-		return nil, errors.NotSupportedf("instance profile creation with %s", awsInstanceProfileAutoCreateVal)
-
-		// Commenting out the below code till the above error is removed.
-
-		//controllerName, ok := args.ControllerConfig[controller.ControllerName].(string)
-		//if !ok {
-		//	return nil, errors.NewNotValid(nil, "cannot find controller name in config")
-		//}
-		//controllerUUID := args.ControllerConfig[controller.ControllerUUIDKey].(string)
-		//instProfile, err := ensureControllerInstanceProfile(
-		//	ctx.Context(),
-		//	e.iamClient,
-		//	controllerName,
-		//	controllerUUID)
-		//if err != nil {
-		//	return nil, err
-		//}
-		//args.BootstrapConstraints.InstanceRole = instProfile.InstanceProfileName
+		controllerName, ok := args.ControllerConfig[controller.ControllerName].(string)
```

Comment:
This feels like it should actually check the map access not the type assertion, same below.

Reply:
well maybe even check the type assertion too